src/main/java/core/basesyntax/service/ReportService.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.activity.ActivityHandler;
+import core.basesyntax.strategy.ActivityStrategy;
+import java.util.List;
+
+public class ReportService {
+    private ActivityStrategy activityStrategy;
+    private FruitDao fruitDao;
+
+    public ReportService(ActivityStrategy activityStrategy) {
+        this.activityStrategy = activityStrategy;
+        fruitDao = new FruitDaoImpl();
+    }
+
+    public boolean executeOperations(List<FruitTransaction> fruitTransactionList) {
+        for (FruitTransaction fruitTransaction : fruitTransactionList) {
+            ActivityHandler activityHandler = activityStrategy.get(fruitTransaction.getOperation());
+            activityHandler.operate(fruitTransaction);
+        }
+        return true;
+    }
+
+    public String generateReport() {
+        return fruitDao.createReport();
+    }
```

Comment:
fruitDao shouldn't be bothered with report creation logic, it should only provide entrySet or the data you need, the logic should be here, in ReportService